src/conversion.jl

```diff
@@ -159,6 +178,9 @@ function Base.convert(::Type{AbstractMatrix}, L::KroneckerSumMap)
     IB = Diagonal(ones(Bool, size(B, 1)))
     return kron(convert(AbstractMatrix, A), IB) + kron(IA, convert(AbstractMatrix, B))
 end
+# seems like code duplication:
+# convert(AbstractMatrix, L) will anyway fall back to Matrix{eltype(L)}(L)
+
```

Comment:
Writing here, as I cannot comment on unchanged lines, but there seems to be an error then in line 188 (new file) or 166 (old line) below, where `convert(AbstractMatrix, A)` should probably be `sparse(A)` or `convert(SparseMatrixCSC, A)` and similar for `B`? Is there a reason to use `convert(SparseMatrixCSC, A)` rather than `sparse(A)` in some of the other sparse conversion methods above?

Reply:
Below, the end result is sparse anyway because `IA` and `IB` are sparse. That suffices to be handled by a `SparseArray.kron` method. Given that, I used the "weakest" conversion method possible, instead of enforcing sparsity on `A` and `B` (which might result in higher memory allocation, when these are `Diagonal`, `Bidiagonal` or stuff like that).

Reply:
Ok, seems like a lot of considerations have gone into all of this which I fail to appreciate by a quick reading.

So maybe one final question: why do several of the methods above use `convert(SparseMatrixCSC, ...)` rather than just `sparse(...)`, e.g. lines  113, 122, 133, 146, 153, 164 ? It seems like defining the `sparse(...)` method is our entry point for this functionality, and we have lines 44 and 45 to handle the case where a user uses a different approach (namely `convert(SparseMatrixCSC, ...)` or just `SparseMatrixCSC(...)`) to do the sparse conversion. But then within the `sparse` definitions, we actually also use a different approach, namely `convert(SparseMatrixCSC, ...)`.

Reply:
I guess this is where you got me. You're right, the root function is `sparse`, which should be overloaded when needed. And by default, this will overload the other `convert` and `SparseMatrixCSC` methods. This is with regards to "the left hand side", the functions to be defined. With regards to the "right hand sides" (how we do it), I use `convert(SparseMatrixCSC, ...)` because it's argument `...` may happen to be sparse already, and in that case, I want to avoid creating a copy. Does that make sense?

Reply:
Ok, and `sparse(A::SparseMatrixCSC)` creates a copy, I see. I think this then settles it all. I will remove my comments and this will then probably be my final commit.